purchase_security/security/security.xml

```diff
@@ -47,5 +65,13 @@
             >['|',('order_id.user_id','=',user.id),('order_id.user_id','=',False)]</field>
             <field name="groups" eval="[(4,ref('group_purchase_own_orders'))]" />
         </record>
+        <record model="ir.rule" id="purchase_order_user_group_purchases_rule">
+            <field name="name">View purchase order lines (purchase responsible)</field>
+            <field name="model_id" ref="purchase.model_purchase_order_line" />
+            <field
+                name="domain_force"
+            >['|', ('order_id.team_id.user_ids', 'in', [user.id]), ('order_id.team_id', '=', False)]</field>
```

Comment:
This is not going to be very performant. What do you think @pedrobaeza ?

Reply:
We can put `auto_join=True` for having less penalty, and it's compatible with security.

Reply:
Thanks for all the suggestions!